### PR TITLE
[EN TasksMode] Merge meal time references with date, if contiguous in text

### PR DESCRIPTION
**Pull request by Neelisha-saxena**
Example: "Take medicine after dinner tomorrow" 
Under Default mode, "after dinner" and "tomorrow" are separate recognition.  But under TasksMode "after dinner tomorrow" is a single recognition.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:224]: This seems too brittle for merge. I'm OK with it to streamline merging your new mode behaviour. But I'd suggest you look at some of the PRs by @aitelint to see how merge was done in other cases.
These specific merges could potentially be applied to the default mode (pending also perf).

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs:68]: Nitpick: Mode this init to the last and add a blank line before it.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs:154]: TasksmodeMealTimeofDayRegex

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs:63]: TasksmodeMealTimeofDayRegex

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs:22]: Same as in previous PR.

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs]: added tasksmode constants.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDateTimePeriodParserConfiguration.cs:68]: changes integrated

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishDateTimePeriodExtractorConfiguration.cs]: renamed variable

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimePeriodExtractorConfiguration.cs]: renamed var

File: .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs:14]: Same comment as in previous PR.

**Review by tellarin — CHANGES_REQUESTED** [review]
This PR itself LGTM, but pending a fix from another PR before it can be rebased and merged.

File: .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs
Comment (Neelisha-saxena) [reply .NET/Microsoft.Recognizers.Text.DateTime/TasksModeConstants.cs:14]: resolved similar to previous PR.

**Review by tellarin — APPROVED** [review]

**Comment by tellarin** [comment]
@Neelisha-saxena Please resolve the conflicts to merge this one first. Thanks.

**Comment by Neelisha-saxena** [comment]
> @Neelisha-saxena Please resolve the conflicts to merge this one first. Thanks.

resolved merge conflicts.

**Review by tellarin — APPROVED** [review]

**Pull request merged**
